var/spack/repos/spack_repo/builtin/packages/llvm/package.py

```diff
@@ -805,26 +805,48 @@ def determine_version(cls, exe):
     def determine_variants(cls, exes, version_str):
         # Do not need to reuse more general logic from CompilerPackage
         # because LLVM has kindly named compilers
-        variants, compilers = {"+clang"}, {}
-        lld_found, lldb_found = False, False
-        for exe in sorted(exes, key=len):
-            name = os.path.basename(exe)
-            if "clang++" in name:
-                compilers.setdefault("cxx", exe)
-            elif "clang" in name:
-                compilers.setdefault("c", exe)
-            elif "flang" in name and "fortran" not in compilers:
-                variants.add("+flang")
-                compilers.setdefault("fortran", exe)
-            elif "ld.lld" in name:
-                lld_found = True
-            elif "lldb" in name:
-                lldb_found = True
-
-        variants.add("+lld" if lld_found else "~lld")
-        variants.add("+lldb" if lldb_found else "~lldb")
-
-        return "".join(sorted(variants)), {"compilers": compilers}
+
+        # Map between exectuable name, variant name, and compiler language.
+        # The ordering of this list is the ordering of the returned variants.
+        exe_variant_lang = [
+            ("clang++", "clang", "cxx"),
+            ("clang", "clang", "c"),
+            ("flang", "flang", "fortran"),
+            # TODO: gold?
```

Comment:
This mostly looks good, but take this TODO out. The ld.gold linker should never be installed as part of the LLVM install.

Reply:
Got it. Thanks!